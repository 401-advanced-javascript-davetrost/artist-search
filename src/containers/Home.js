import React from 'react';
import { useSearch } from './use-search';
import { usePaging } from './use-paging';
import { useArtistSearch } from './use-artist-search';

import { MemoArtistDeck } from './Artist-Deck';
import Search from '../components/Search';
import styles from './Home.css';

export default function Home() {
  const { search, handleChange } = useSearch();
  const { page, handleBack, handleNext } = usePaging();
  const { artists, handleSubmit } = useArtistSearch(page, search);
  
  return (
    <div className={styles.Home}>
      <Search
        handleSubmit={handleSubmit}
        handleChange={handleChange} 
      />
      <MemoArtistDeck
        artists={artists}
        handleBack={handleBack}
        handleNext={handleNext}
      />
    </div>
  );
}
