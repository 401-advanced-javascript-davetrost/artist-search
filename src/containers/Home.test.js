import React from 'react';
import { shallow } from 'enzyme';
import Home from './Home';

import useArtistSearchMock from './use-artist-search';
jest.mock('./use-artist-search', () => ({}));

describe('Home component', () => {

  it('renders the initial state', () => {
    useArtistSearchMock.useArtistSearch = () => ({ 
      data: false, 
      handleSubmit: () => true,
      handleChange: () => true,
    });
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a list of artists', () => {
    useArtistSearchMock.useArtistSearch = () => ({ 
      data: true, 
      artists: ['ABBA', 'Beatles', 'Led Zep', 'MIA'],
      handleSubmit: () => true,
      handleChange: () => true,
    });
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });
});
