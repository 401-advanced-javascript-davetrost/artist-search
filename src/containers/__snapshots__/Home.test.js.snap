// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders a list of artists 1`] = `
<div
  className="Home"
>
  <Search
    handleChange={[Function]}
    handleSubmit={[Function]}
  />
  <Memo(ArtistDeck)
    artists={
      Array [
        "ABBA",
        "Beatles",
        "Led Zep",
        "MIA",
      ]
    }
    handleBack={[Function]}
    handleNext={[Function]}
  />
</div>
`;

exports[`Home component renders the initial state 1`] = `
<div
  className="Home"
>
  <Search
    handleChange={[Function]}
    handleSubmit={[Function]}
  />
</div>
`;
